### DISCUSSION TITLE
FFMPEG6 + PyAV 13.0.0

### DISCUSSION THREAD

**Discussion by hmaarrfk** [PyAV-Org/PyAV#1518] [4. Bugs]
I seem to be getting some failures in tests on linux+x86 (should be easily testable)
when using FFMPEG6 + PyAV 13.0.0

https://github.com/conda-forge/av-feedstock/pull/80

I started a new discussion since I don't want to take away from the monumental celegration that 13.0.0 (FFMPEG7 support is huge!!!!)
https://github.com/PyAV-Org/PyAV/discussions/1508

In particular the failure seems to be:

```
=================================== FAILURES ===================================
_______________ TestAudioResampler.test_pts_assertion_same_rate ________________

self = <tests.test_audioresampler.TestAudioResampler testMethod=test_pts_assertion_same_rate>

    def test_pts_assertion_same_rate(self):
        import av
    
        av.logging.set_level(av.logging.VERBOSE)
        resampler = AudioResampler("s16", "mono")
    
        # resample one frame
        iframe = AudioFrame("s16", "stereo", 1024)
        iframe.sample_rate = 48000
        iframe.time_base = Fraction(1, 48000)
        iframe.pts = 0
    
        oframes = resampler.resample(iframe)
        self.assertEqual(len(oframes), 1)
    
        oframe = oframes[0]
        self.assertEqual(oframe.pts, 0)
        self.assertEqual(oframe.time_base, iframe.time_base)
        self.assertEqual(oframe.sample_rate, iframe.sample_rate)
        self.assertEqual(oframe.samples, iframe.samples)
    
        # resample another frame
        iframe.pts = 1024
    
>       oframes = resampler.resample(iframe)

tests/test_audioresampler.py:95: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
av/audio/resampler.pyx:33: in av.audio.resampler.AudioResampler.resample
    ???
av/audio/resampler.pyx:106: in av.audio.resampler.AudioResampler.resample
    ???
av/filter/graph.pyx:198: in av.filter.graph.Graph.push
    ???
av/filter/context.pyx:96: in av.filter.context.FilterContext.push
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   av.error.ValueError: [Errno 22] Invalid argument; last error log: [abuffer] Layout indicates a different number of channels than actually present

av/error.pyx:326: ValueError
------------------------------ Captured log call -------------------------------
ERROR    libav.abuffer:test_audioresampler.py:95 Layout indicates a different number of channels than actually present
=========================== short test summary info ============================
```

full build log (sorry using conda(-forge))
[206.fail.txt](https://github.com/user-attachments/files/16882396/206.fail.txt)

**Comment by hmaarrfk** [comment]
Don't get me wrong, i'm more than happy to drop FFMPEG6 for this version (it might keep things simpler) but I know @jlaine was quite particular about maintaining some broad compatibility matrix with FFMPEGs.

**Comment by hmaarrfk** [comment]
It seems maybe i don't know much about audio. Tips on what might be wrong with the conda-forge environment are appreciated.

**Comment by WyattBlue** [comment]
When you're using ffmpeg 6. You need to run `python scripts/comptime.py 6` before building

**Reply by hmaarrfk** [reply]
Do i need something similar for 7?

**Reply by WyattBlue** [reply]
No, ffmpeg 7 works right out of the box.

**Reply by hmaarrfk** [reply]
Ok thanks for the tip will try.

BTW, with Cython, there are ways to do these kinds of "preprocessor" definitions. I had to do them when exposing some C features for HDF5.

the "DIRECT" feature was conditionally enabled with an IF statement that was "compiled in" so no loss of performance at runtime depending on what was availble at compile time.

https://github.com/h5py/h5py/commit/a82f1e8ab2a07089967a08e16c9cb1db73252c22

you might be able to find the C-Define that specifies the version of LibAV that you can "switch on".
